plugins/ingest/src/test/java/org/elasticsearch/ingest/IngestClientIT.java

```diff
@@ -55,6 +59,58 @@
 
     }
 
+    public void testSimulate() throws Exception {
+        new PutPipelineRequestBuilder(client(), PutPipelineAction.INSTANCE)
+                .setId("_id")
+                .setSource(jsonBuilder().startObject()
+                        .field("description", "my_pipeline")
+                        .startArray("processors")
+                        .startObject()
+                        .startObject("grok")
+                        .field("field", "field1")
+                        .field("pattern", "%{NUMBER:val:float} %{NUMBER:status:int} <%{WORD:msg}>")
+                        .endObject()
+                        .endObject()
+                        .endArray()
+                        .endObject().bytes())
+                .get();
+        assertBusy(new Runnable() {
+            @Override
+            public void run() {
+                GetPipelineResponse response = new GetPipelineRequestBuilder(client(), GetPipelineAction.INSTANCE)
+                        .setIds("_id")
+                        .get();
+                assertThat(response.isFound(), is(true));
+                assertThat(response.pipelines().get("_id"), notNullValue());
+            }
+        });
+
+        SimulatePipelineResponse response = new SimulatePipelineRequestBuilder(client(), SimulatePipelineAction.INSTANCE)
+                .setId("_id")
+                .setSource(jsonBuilder().startObject()
+                        .startArray("docs")
+                        .startObject()
+                        .field("_index", "index")
+                        .field("_type", "type")
+                        .field("_id", "id")
+                        .startObject("_source")
+                        .field("foo", "bar")
+                        .endObject()
+                        .endObject()
+                        .endArray()
+                        .endObject().bytes())
+                .get();
+
+        Map<String, Object> expectedDoc = new HashMap<>();
+        expectedDoc.put("foo", "bar");
+        Data expectedData = new Data("index", "type", "id", expectedDoc);
+        SimulateDocumentResult expectedResponse = new SimulateDocumentResult(expectedData);
+        List<SimulateDocumentResult> expectedResponses = Arrays.asList(expectedResponse);
+        SimulatePipelineResponse expected = new SimulatePipelineResponse("_id", expectedResponses);
+
+        assertThat(response, equalTo(expected));
+    }
+
```

Comment:
I don't see yet specific equals/hashcode tests, I think we could benefit from those. I know that they have been generated and are most likely fine at the moment, but pretty sure without tests they will get out of sync with future modifications.